contracts/interfaces/ICover.sol

```diff
@@ -9,27 +9,44 @@ interface ICover is IERC721 {
   /* ========== DATA STRUCTURES ========== */
 
   struct StakingPool {
-    uint160 id;
-    uint96 bookedAmount;
+    address poolAddress;
+    uint96 coverAmount;
+    uint96 premiumInNXM;
   }
 
   struct Cover {
     uint24 productId;
     uint8 payoutAsset;
-    uint8 deniedClaims;
     uint96 amount;
     uint32 start;
     uint32 period;  // seconds
+    uint96 premium;
```

Comment:
Keeping in mind that we already have the premium in NXM in the pools array, do we still need the premium in asset here? We could just emit an event with the paid amount if we want to calculate something off-chain, but if we're not using it anywhere on-chain - let's delete it. One less write, and would fit the cover in a single slot!

Reply:
@shark0der  i use this to do calculations for refunds, remaining premiums etc.

https://github.com/NexusMutual/smart-contracts/blob/292ec224f6cc8a98b83ec94b75cb5783a2eed80c/contracts/modules/cover/Cover.sol#L318

this is one location